codex-rs/cli/src/mcp_cmd.rs

```diff
@@ -201,15 +204,33 @@ fn run_list(config_overrides: &CliConfigOverrides, list_args: ListArgs) -> Resul
         let json_entries: Vec<_> = entries
             .into_iter()
             .map(|(name, cfg)| {
-                let env = cfg.env.as_ref().map(|env| {
-                    env.iter()
-                        .map(|(k, v)| (k.clone(), v.clone()))
-                        .collect::<BTreeMap<_, _>>()
-                });
+                let env = match &cfg.transport {
+                    McpServerTransportConfig::Stdio { env, .. } => env.as_ref().map(|env| {
+                        env.iter()
+                            .map(|(k, v)| (k.clone(), v.clone()))
+                            .collect::<BTreeMap<_, _>>()
+                    }),
+                    McpServerTransportConfig::StreamableHttp { .. } => None,
+                };
+                let transport = match &cfg.transport {
```

Comment:
Why `match &cfg.transport` twice? You only use the `env` in the `Stdio` case.

Reply:
Oh this whole env block can be removed because it's embedded in transport